clients/rospy/src/rospy/impl/masterslave.py

```diff
@@ -441,13 +442,19 @@ def _connect_topic(self, topic, pub_uri):
         interval = 0.5  # seconds
         # while the ROS node is not shutdown try to get the topic information
         # and retry on connections problems after some wait
+        # Abort the retry if the we get a Connection Refused since at that point
+        # we know for sure the URI is invalid
         while not success and not is_shutdown():
             try:
                 code, msg, result = \
                       xmlrpcapi(pub_uri).requestTopic(caller_id, topic, protocols)
                 success = True
             except Exception as e:
-                if not is_shutdown():
+                if getattr(e, 'errno', None) == errno.ECONNREFUSED:
+                    code = errno.ECONNREFUSED
+                    msg = str(e)
+                    break

```

Comment:
The early return down below only triggers when `code <= 0`, but you're assigning it a positive value. What is the expected behaviour here, when we break out of the while loop?